packages/core/src/Select/Select.stories.tsx

```diff
@@ -0,0 +1,183 @@
+    ),
+  ],
+  render: () => {
+    return (
+      <>
+        <HvSelect required label="Required" />
+        <HvSelect disabled label="Disabled" />
+        <HvSelect readOnly label="Read-only" />
+        <HvSelect status="invalid" label="Invalid" />
+      </>
+    );
+  },
+};
+
+export const Form: StoryObj<HvSelectProps<{}, false>> = {
+  parameters: {
+    docs: {
+      description: {
+        story:
+          "To integrate `HvSelect` in a form, make sure you're giving it a `name`. <br />\
+          The value result will be the selected option's `value`, or a JSON of the selected values when multi-select is enabled. The value can be customized via the `getSerializedValue` prop.",
+      },
+    },
+  },
+  decorators: [decorator],
+  render: () => {
+    return (
+      <form
+        onSubmit={(evt) => {
+          evt.preventDefault();
+          const formData = new FormData(evt.currentTarget);
+          alert(JSON.stringify(Object.fromEntries(formData)));
+        }}
+      >
+        <HvSelect
+          multiple
+          required
+          name="countries"
+          label="Country"
+          description="Select your favorite countries"
+          placeholder="Select countries"
+          getSerializedValue={(values) => values.map((v) => v.value).join(",")}
+          onChange={(evt, val) => console.log(val)}
+        >
+          {options.map(({ value, label, flag }) => (
+            <HvOption key={value} value={value} label={label}>
+              {`${flag} ${label}`}
+            </HvOption>
+          ))}
+        </HvSelect>
+        <br />
+        <HvButton type="submit" variant="secondarySubtle">
+          Submit
+        </HvButton>
+      </form>
+    );
+  },
+};
+
+export const Controlled: StoryObj<HvSelectProps<{}, false>> = {
+  parameters: {
+    docs: {
+      description: {
+        story:
+          "The value and open states of `HvSelect` can be controlled by using the `value`/`onChange` and `open`/`onOpenChange` props respectively.",
+      },
+    },
+  },
+  decorators: [
+    (Story) => <div className="flex gap-sm min-h-[300px]">{Story()}</div>,
+  ],
+  render: () => {
+    const [selection, setSelection] = useState<string[]>([]);
+    const [open, setOpen] = useState(false);
+
+    const anySelected = selection.length > 0;
+
+    return (
+      <>
+        <HvSelect
+          multiple
+          required
+          open={open}
+          name="countries"
+          aria-label="Country"
+          placeholder="Select countries"
+          value={selection}
+          style={{ width: 300 }}
+          onChange={(evt, val) => setSelection(val)}
+          onOpenChange={setOpen}
+        >
+          {options.map(({ value, label, flag }) => (
+            <HvOption key={value} value={value} label={label}>
+              {`${flag} ${label}`}
+            </HvOption>
+          ))}
+        </HvSelect>
+        <HvButton variant="secondarySubtle" onClick={() => setOpen((o) => !o)}>
```

Comment:
There's a weird behaviour here: if a click on the button to close the dropdown, it closes and opens again.

Reply:
The sample is both controlled and uncontrolled, that's why. I'll remove the button, but keep it controlled.
I'll also move this sample to a separate file to `?raw` load the source code